File: IDATT2106 Systemutvikling 2 med smidig prosjekt/backend/src/main/java/com/ntnu/gidd/service/User/UserService.java
package com.ntnu.gidd.service.User;

import com.ntnu.gidd.dto.User.UserDto;
import com.ntnu.gidd.dto.User.UserPasswordResetDto;
import com.ntnu.gidd.dto.User.UserPasswordUpdateDto;
import com.ntnu.gidd.dto.User.UserRegistrationDto;
import com.ntnu.gidd.model.User;
import com.querydsl.core.types.Predicate;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import com.ntnu.gidd.model.PasswordResetToken;
import org.springframework.stereotype.Service;

import javax.mail.MessagingException;
import java.security.Principal;
import java.util.UUID;

@Service
public interface UserService {
      UserDto updateUser(UUID id, UserDto user);
      UserDto getUserDtoByEmail(String email);
      UserDto saveUser(UserRegistrationDto user);
      UserDto deleteUser(String email);
      void changePassword(Principal principal, UserPasswordUpdateDto user);
      Page<UserDto> getAll(Predicate predicate, Pageable pageable);
      UserDto getUserByUUID(UUID userId);
      void forgotPassword(String email) throws MessagingException;
      void validateResetPassword(UserPasswordResetDto user, UUID passwordResetTokenId);

      User getUserById(UUID actorId);
}

File: IDATT2105 Full-stack applikasjonsutvikling/Rombestilling/backend/src/main/kotlin/ntnu/idatt2105/user/service/UserService.kt
package ntnu.idatt2105.user.service

import com.querydsl.core.types.Predicate
import ntnu.idatt2105.core.response.Response
import ntnu.idatt2105.reservation.service.ReserverService
import ntnu.idatt2105.security.dto.ForgotPassword
import ntnu.idatt2105.security.dto.MakeAdminDto
import ntnu.idatt2105.security.dto.ResetPasswordDto
import ntnu.idatt2105.user.dto.DetailedUserDto
import ntnu.idatt2105.user.dto.UserDto
import ntnu.idatt2105.user.dto.UserRegistrationDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.web.multipart.MultipartFile
import java.util.*

interface UserService : ReserverService {
    fun registerUser(user: UserRegistrationDto): UserDto
    fun registerUserBatch(file: MultipartFile): Response
    fun updateUser(id: UUID, user: UserDto): UserDto
    fun getUsers(pageable: Pageable, predicate: Predicate): Page<UserDto>
    fun forgotPassword(email: ForgotPassword)
    fun resetPassword(resetDto: ResetPasswordDto, id: UUID)
    fun <T> getUser(id: UUID, mapTo: Class<T>): T
    fun deleteUser(id: UUID): Response
    fun makeAdmin(user: MakeAdminDto): DetailedUserDto
    fun invalidateExpiredUsers()
}
